📦 NEW: Add filter to show hidden posts only

nielslange opened the pull request:
Fixes #26 

**Change**

Add filter to the WP Admin post page to allow users to show hidden posts only.

**Steps to test**

1. Check out this plugin
2. Go to `/wp-admin/edit.php`
3. Create at least one hidden post
4. Look up `/wp-admin/edit.php` again

<table>
<tr>
<td>Before:
<br><br>

![#26-before](https://user-images.githubusercontent.com/3323310/111898849-0d9d4600-8a5b-11eb-8997-4e109072c88e.png)
</td>
<td>After:
<br><br>

![#26-after](https://user-images.githubusercontent.com/3323310/111898850-0fffa000-8a5b-11eb-954c-a5c28524739e.png)
</td>
</tr>
</table> 

**Note**

- This PR depends on PR ~#27~ #31 .
- In lines 197, 211 & 219, I used `//phpcs:ignore` to suppress the warning regarding a missing nonce, which is not needed in this case.
- In line 220, I used `// phpcs:ignore` to suppress an error regarding overwriting the `$wp_query` object, which is needed in this case.

nielslange committed:
📦 NEW: Add filter to show hidden posts only

nielslange committed:
🐛 FIX: Correct wrong type declaration

nielslange commented:
@trepmal Would you mind reviewing this PR once #31 had been merged?

trepmal commented on hidden-posts.php:
```diff
@@ -180,6 +181,48 @@ public function custom_column_style() {
+			return $views;
+		}
+
+		if ( ! count( get_option( 'hidden-posts' ) ) ) {
```
I think this (and the next instance) need to set a default value, or else if no posts have yet been hidden, it will throw a warning about an uncountable value.

nielslange replied:
Thanks for pointing this out. When testing, I did not see a warning by default, as the call `get_option( 'hidden-posts' )` returns an empty array when no hidden posts exist. However, when deleting the option `hidden-posts` directly within  the database, the following warning appears:

```
( ! ) Warning: count(): Parameter must be an array or an object that implements Countable in /Users/nielslange/Development/plugins/hidden-posts/hidden-posts.php on line 201
```

As suggested by you, I've added a default value for both `get_option()` function calls to prevent a warning from being thrown.

trepmal commented on hidden-posts.php:
```diff
@@ -180,6 +181,48 @@ public function custom_column_style() {
+			return $views;
+		}
+
+		if ( isset( $_GET['post_type'] ) && 'post' !== $_GET['post_type'] ) { // phpcs:ignore
```
I'd like to see the `phpcs:ignore` specify which rule it's ignoring.

nielslange replied:
Good catch, @trepmal. Until now, I've only used `// phpcs:ignore` without further explanation. Now that you pointed this out, I see the benefit of adding the ignored rule to it for transparency reasons.

In the particular case that you highlighted, I added the `// phpcs:ignore` statement because of a missing nonce. As the related function only reads data, but do not write and/or edit data, I decided to ignore the missing nonce.

nielslange committed:
Add context to phpcs:ignore statements

nielslange committed:
Add default value to get_option() statement to prevent warnings

nielslange committed:
Move variable $class out of sprintf() format section

nielslange commented:
Thanks for reviewing this PR, @trepmal, and requesting the changes. I just addressed them and added some notes. Please let me know if you spot anything else that should be changed.

trepmal commented on hidden-posts.php:
```diff
@@ -180,6 +181,49 @@ public function custom_column_style() {
+			return $views;
+		}
+
+		if ( ! count( get_option( self::META_KEY, array() ) ) ) {
```
I know you didn't introduce it here, but we should consider renaming this constant since we're not dealing with meta.

nielslange replied:
Great suggestion, @trepmal. I just renamed `META_KEY` to `OPTION_KEY` to reflect the table that key is stored in.

trepmal reviewed (approved):
Looks good!

nielslange committed:
Rename META_KEY to OPTION_KEY to reflect storage location
